Add base API crawler class

aperrin66 opened the pull request:
This is preparation to adding support for Creodias (#57).
The code which is common to paginated API crawlers has been moved to a new `HTTPPaginatedAPICrawler` class.

opsdep commented on tests/test_crawlers.py:
```diff
@@ -561,90 +561,98 @@ def test_get_download_url_no_direct_download_link(self, mock_get_link, mock_http
+class HTTPPaginatedAPICrawlerTestCase(unittest.TestCase):
+    """Tests for the HTTPPaginatedAPICrawler base class"""
+
+    class TestCrawler(crawlers.HTTPPaginatedAPICrawler):
```
A class inside a class must be in the setup method in order to be fresh and unchanged of execution of the other tests.

aperrin66 replied:
This is a class definition, not an instantiation.
It is instantiated in the `setUp()` method.

opsdep commented on tests/test_crawlers.py:
```diff
@@ -561,90 +561,98 @@ def test_get_download_url_no_direct_download_link(self, mock_get_link, mock_http
-                'start': 0,
-                'rows': self.PAGE_SIZE,
-                'orderby': 'beginposition asc'
-            },
-            'auth': ('user', 'pass')
-        })
-        self.assertEqual(self.crawler._urls, [])
-
     def test_build_parameters_with_standard_time_range(self):
         """Build the request parameters with a time range composed of two datetime objects"""
-        request_parameters = crawlers.CopernicusOpenSearchAPICrawler._build_request_parameters(
+        request_parameters = self.crawler._build_request_parameters(
```
You define a class inside another class in a nested manner. Please do it in the setup method of tests in order to prevent tests and actions inside each test manipulating other tests and their actions.

aperrin66 replied:
There is no nested class in the `CopernicusOpenSearchAPICrawlerTestCase`, and the `self.crawler` attribute used on this line is defined in the `setUp()` method.

aanersc reviewed (approved):
Can you explain what is the difference between the test file for this PR? You have removed two data test file and some modifications in the test

aperrin66 commented:
> Can you explain what is the difference between the test file for this PR? You have removed two data test file and some modifications in the test

I moved common functionalities from the Copernicus crawler to the HTTPPaginatedAPICrawler base class, and I did the same in the tests: I moved the tests dealing with functionalities from the base class to a new TestCase. In so doing, some of the test data became useless, which is why the files were removed.
